src/main/java/tr/edu/itu/cavabunga/client/http/HttpAdapter.java

```diff
@@ -0,0 +1,49 @@
+package tr.edu.itu.cavabunga.client.http;
+
+import tr.edu.itu.cavabunga.client.exception.ClientException;
+import lombok.Data;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@Data
+@Component
+public class HttpAdapter {
+    public static String doRequest(String requestUrl, RequestMethod requestMethod, String bodyParameters){
```

Comment:
why do not we use generic rest template for spring applications: https://spring.io/guides/gs/consuming-rest/ ? It can be beneficial for further development.